docs/design/upstream-datamover.md

```diff
@@ -0,0 +1,110 @@
+# Design for Integrating Velero Built-in Data Mover (VBDM) support into OADP
+
+
+## Abstract
+One of the primary new features of Velero 1.12 is the new Velero Built-in Data Mover (VBDM).
+The OADP Operator needs to add configuration (and e2e tests) to support this new datamover and remove any configuration and testing of the legacy OADP datamover.
+
+## Background
+Prior to the release of Velero 1.12, OADP 1.1 and 1.2 integrated an OADP-specific datamover based on Volsync into OADP and the OADP-installed Velero version.
+This took the form of some modifications to Velero core and the Velero CSI plugins, a new VSM plugin, and a separate VSM controller that used Volsync to store Volume data in the BackupStorageLocation.
+In addition to this, DPA and installation support for the OADP datamover was included in the OADP operator.
+
+In order to integrate VBDM into OADP, all of the above legacy OADP datamover parts will be removed from the OADP operator, the OADP Velero build, and the OADP build of the Velero Plugin for CSI.
+
```

Comment:
Maybe also add a link regarding the Node Agent Configuration design as a dependency for this design ?

Reply:
@shubham-pampattiwar Technically it's not a dependency. That's why I didn't really reference its contents. Everything in this design works just fine enabling the legacy restic config.